cmd/account/rotate-secret.go

```diff
@@ -80,128 +57,171 @@ func newRotateSecretOptions(streams genericclioptions.IOStreams, flags *genericc
+	// Username is osdManagedAdmin-aaabbb
+	osdManagedAdminUsername := common.OSDManagedAdminIAM + "-" + accountIDSuffixLabel
+
+	createAccessKeyOutput, err := awsClient.CreateAccessKey(&iam.CreateAccessKeyInput{
+		UserName: aws.String(osdManagedAdminUsername),
+	})
 	if err != nil {
 		return err
 	}
 
-	// the specified user not exist, create one
-	if !ok {
-		policyArn := aws.String("arn:aws:iam::aws:policy/AdministratorAccess")
-		if err := awsprovider.CreateIAMUserAndAttachPolicy(awsClient,
-			username, policyArn); err != nil {
-			return err
-		}
-	} else {
-		fmt.Fprintf(o.IOStreams.Out, "User %s exists, deleting existing access keys now.\n", o.iamUsername)
-		if err := awsprovider.DeleteUserAccessKeys(awsClient, username); err != nil {
-			return err
-		}
+	// Place new credentials into body for secret
+	newOsdManagedAdminSecretData := map[string][]byte{
+		"aws_user_name":         []byte(*createAccessKeyOutput.AccessKey.UserName),
+		"aws_access_key_id":     []byte(*createAccessKeyOutput.AccessKey.AccessKeyId),
+		"aws_secret_access_key": []byte(*createAccessKeyOutput.AccessKey.SecretAccessKey),
+	}
+
+	// Update existing osdManagedAdmin secret
+	err = common.UpdateSecret(o.kubeCli, o.accountCRName+"-secret", common.AWSAccountNamespace, newOsdManagedAdminSecretData)
+	if err != nil {
+		return err
 	}
 
-	newKey, err := awsClient.CreateAccessKey(&iam.CreateAccessKeyInput{
-		UserName: username,
+	// Update secret in ClusterDeployment's namespace
+	err = common.UpdateSecret(o.kubeCli, "aws", account.Spec.ClaimLinkNamespace, newOsdManagedAdminSecretData)
+	if err != nil {
+		return err
+	}
+
+	// Update osdManagedAdminSRE secret
+	// Username is osdManagedAdminSRE-aaabbb
+	osdManagedAdminSREUsername := common.OSDManagedAdminIAM + "SRE" + "-" + accountIDSuffixLabel
+	createAccessKeyOutputSRE, err := awsClient.CreateAccessKey(&iam.CreateAccessKeyInput{
+		UserName: aws.String(osdManagedAdminSREUsername),
 	})
 	if err != nil {
 		return err
 	}
 
-	secret := k8s.NewAWSSecret(
-		o.secretName,
-		o.secretNamespace,
-		*newKey.AccessKey.AccessKeyId,
-		*newKey.AccessKey.SecretAccessKey,
-	)
-	if o.printSecret {
-		fmt.Fprintln(o.IOStreams.Out, secret)
+	// Place new credentials into body for secret
+	newOsdManagedAdminSRESecretData := map[string][]byte{
+		"aws_user_name":         []byte(*createAccessKeyOutputSRE.AccessKey.UserName),
+		"aws_access_key_id":     []byte(*createAccessKeyOutputSRE.AccessKey.AccessKeyId),
+		"aws_secret_access_key": []byte(*createAccessKeyOutputSRE.AccessKey.SecretAccessKey),
 	}
 
-	if o.outputPath != "" {
-		outputPath, err := filepath.Abs(o.outputPath)
-		if err != nil {
-			return err
+	// Update existing osdManagedAdmin secret
+	err = common.UpdateSecret(o.kubeCli, o.accountCRName+"-osdManagedAdminSRE-secret", common.AWSAccountNamespace, newOsdManagedAdminSRESecretData)
+	if err != nil {
+		return err
+	}
+
+	// Only update osdCcsAdmin credential if specified
+	if o.osdCcsAdmin {
+		if account.Spec.BYOC {
```

Comment:
In case the `--ccs` argument has been passed we should output a confirmation, indicating we've successfully updated the CCS creds

Reply:
👍